xgcm/test/test_metrics.py

```diff
@@ -311,28 +307,86 @@ def test_set_metric():
         assert k in grid_manual._metrics.keys()
 
         for metric_expected, metric in zip(v, grid_manual._metrics[k]):
-            xr.testing.assert_equal(metric_expected.reset_coords(drop=True), metric)
+            xr.testing.assert_allclose(metric_expected.reset_coords(drop=True), metric)
 
         for metric_expected, metric in zip(v, grid._metrics[k]):
-            xr.testing.assert_equal(metric_expected.reset_coords(drop=True), metric)
+            xr.testing.assert_allclose(metric_expected.reset_coords(drop=True), metric)
 
 
-@pytest.mark.skip(reason="For the next PR")
 @pytest.mark.parametrize(
-    "metric_axes,metric_name",
+    "metric_axes, metric_varname, expected_varname",
     [
-        ("X", "dx_t"),
-        ("Y", "dy_ne"),
+        ("X", "dx_n", ["dx_t", "dx_n_overwrite", "dx_e", "dx_ne"]),
+        (
+            ("Y", "X"),
+            "area_n",
+            ["area_t", "area_n_overwrite", "area_e", "area_ne"],
+        ),
     ],
 )
-def test_interp_metrics(metric_axes, metric_name):
-    # need to test with boundary and fill_value conditions
+def test_set_metric_overwrite_true(metric_axes, metric_varname, expected_varname):
+    ds, coords, metrics = datasets_grid_metric("C")
+
+    mv_modified = f"{metric_varname}_overwrite"
+    ds = ds.assign_coords({mv_modified: ds[metric_varname] * 10})
+
+    grid = Grid(ds, coords=coords, metrics=metrics)
+
+    grid.set_metrics(metric_axes, mv_modified, overwrite=True)
+
+    if len(metric_axes) > 1:
+        a, b = metric_axes
+        key = frozenset({a, b})
+    else:
+        key = frozenset({metric_axes})
+
+    set_metric = grid._metrics.get(key)
+
+    expected_metric = []
+    for ev in expected_varname:
+        metric_var = ds[ev].reset_coords(drop=True)
+        expected_metric.append(metric_var)
+
+    assert len(set_metric) == len(expected_metric)
+
+    for i in range(len(set_metric)):
+        assert set_metric[i].equals(expected_metric[i])
+
+
+def test_set_metric_overwrite_false():
+    ds, coords, metrics = datasets_grid_metric("C")
+
+    ds = ds.assign_coords({"dx_t_overwrite": ds["dx_t"] * 10})
+
+    grid = Grid(ds, coords=coords, metrics=metrics)
+
+    with pytest.raises(ValueError, match="setting overwrite=True."):
+        grid.set_metrics("X", "dx_t_overwrite", overwrite=False)
+
+
+def test_set_metric_replace():
```

Comment:
I couldn't integrate this test with `test_set_metric_replace` because if metrics are set as `metrics=metrics` from the beginning, I can't append a new metric when its dimensions already matches an existing metric (can only overwrite).

Reply:
You dont have to set all the metrics in the beginning. How about you modify the `metrics` input dict according to your input parameters:
```python
metric_names = ['dx_t', 'dx_ne'] #not sure this is the right variable name, just a sketch
ds, coords, metrics = ....
metrics = {k:[m for m in v if m in metric_names] for k,v in metrics.items()} # this will then only have the above names in it.
grid = Grid(..., metrics=metrics, ...)
```